src/check_release.rs

```diff
@@ -113,6 +114,7 @@ pub(super) fn run_check_release(
     let queries_to_run: Vec<_> = queries
         .iter()
         .filter(|(_, query)| !version_change.supports_requirement(query.required_update))
+        .filter(|(_, query)| query.relevant_at_lint_level(lint_level))
```

Comment:
Don't these two checks to the same thing, just from different sources?  I feel like the lint_level flag should be `Option<LintLevel>` and if its `None`, then in the above code we use the `version_change` logic to set it. We then only need one `filter` on the query to perform.

Reply:
I thought this was the whole point. We don't want to override it, only filter it. If the version changed by a major increment, we don't want to run minor checks no matter what the lint level is set to.

Reply:
You are proposing pretty much exactly the change I had earlier (`--assume-semver`) which was rejected for that reason.

Reply:
This is being further discussed in the issue